test failure looks legit:
```py
19:56:37 FAIL: test_set_grad_enabled (test_autograd.TestAutograd)
19:56:37 ----------------------------------------------------------------------
19:56:37 Traceback (most recent call last):
19:56:37   File "test_autograd.py", line 1812, in test_set_grad_enabled
19:56:37     self.assertTrue(y.requires_grad)
19:56:37 AssertionError: False is not true
```

--- a/test/test_autograd.py
+++ b/test/test_autograd.py
@@ -1796,21 +1796,6 @@ class TestAutograd(TestCase):
             y = x * 2
         self.assertTrue(y.requires_grad)
 
-        @torch.set_grad_enabled(False)
-        def doubler_without(x):
-            return x * 2
-
-        y = doubler_without(x)
-        self.assertFalse(y.requires_grad)
-
-        @torch.set_grad_enabled(True)
-        def doubler_with(x):
-            return x * 2
-
-        torch.set_grad_enabled(False)
-        y = doubler_with(x)
-        self.assertTrue(y.requires_grad)
-
     def test_reentrant(self):
         y_data = torch.randn(2, 2)
 

--- a/torch/autograd/grad_mode.py
+++ b/torch/autograd/grad_mode.py
@@ -98,7 +98,7 @@ class set_grad_enabled(object):
     r"""Context-manager that sets gradient calculation to on or off.
 
     ``set_grad_enabled`` will enable or disable grads based on its argument :attr:`mode`.
-    It can be used as a context-manager, as a function, or as a decorator.
+    It can be used as a context-manager or as a function.
 
     Arguments:
         mode (bool): Flag whether to enable grad (``True``), or disable
@@ -135,10 +135,3 @@ class set_grad_enabled(object):
     def __exit__(self, *args):
         torch.set_grad_enabled(self.prev)
         return False
-
-    def __call__(self, func):
-        @functools.wraps(func)
-        def decorate_set_grad_enabled(*args, **kwargs):
-            with self:
-                return func(*args, **kwargs)
-        return decorate_set_grad_enabled

